streaming/src/test/scala/org/apache/spark/streaming/ReceivedBlockHandlerSuite.scala

```diff
@@ -113,6 +129,78 @@ class ReceivedBlockHandlerSuite
     }
   }
 
+  test("BlockManagerBasedBlockHandler - count messages MEMORY_ONLY") {
+    handler = makeBlockManagerBasedBlockHandler(blockManager, StorageLevel.MEMORY_ONLY)
+    val block1 = List.fill(100)(new Array[Byte](100))
+    val blockId1 = generateBlockId()
+    val blockStoreResult1 = handler.storeBlock(blockId1, IteratorBlock(block1.iterator))
+    val block2 = collection.mutable.ArrayBuffer.fill(100)(0)
+    val blockId2 = generateBlockId()
+    val blockStoreResult2 = handler.storeBlock(blockId2, ArrayBufferBlock(block2))
+    assert(blockStoreResult1.numRecords === 100)
+    assert(blockStoreResult2.numRecords === 100)
+  }
+
+  test("BlockManagerBasedBlockHandler - count messages DISK_ONLY") {
+    handler = makeBlockManagerBasedBlockHandler(blockManager, StorageLevel.DISK_ONLY)
+    val block1 = List.fill(100)(new Array[Byte](100))
+    val blockId1 = generateBlockId()
+    val blockStoreResult1 = handler.storeBlock(blockId1, IteratorBlock(block1.iterator))
+    val block2 = collection.mutable.ArrayBuffer.fill(100)(0)
+    val blockId2 = generateBlockId()
+    val blockStoreResult2 = handler.storeBlock(blockId2, ArrayBufferBlock(block2))
+    assert(blockStoreResult1.numRecords === 100)
+    assert(blockStoreResult2.numRecords === 100)
+  }
+
+  test("BlockManagerBasedBlockHandler - count messages MEMORY_AND_DISK") {
+    handler = makeBlockManagerBasedBlockHandler(blockManager, StorageLevel.MEMORY_AND_DISK)
```

Comment:
This is repeated code! Cant these be made into a function which take storage level as a parameter? Each of these unit tests then just calls that function with appropriate parameters.